R/pmcmc.R

```diff
@@ -66,6 +65,59 @@ pmcmc <- function(pars, filter, initial = NULL, control = NULL) {
 }
 
 
+##' Run a pMCMC, with sensible random number behaviour, but schedule
+##' execution of the chains yourself. Use this if you want to
+##' distribute chains over (say) the nodes of an HPC system.
+##'
+##' @title pMCMC with manual chain scheduling
+##'
+##' @inheritParams pmcmc
+##' @export
+pmcmc_chains_prepare <- function(pars, filter, initial = NULL, control = NULL) {
+  assert_is(pars, c("pmcmc_parameters", "pmcmc_parameters_nested"))
+  assert_is(filter, "particle_filter")
+  assert_is(control, "pmcmc_control")
+
+  if (control$n_workers != 1) {
+    stop("'n_workers' must be 1")
```

Comment:
Might be nice if this was explained in minimal detail in the document, seems counter-intuitive unless you know how it functions immediately